### PR TITLE
Fix stub generation, change graph exporting for arrows to go to outputs

### LINKED ISSUE #454: [BUG] successfully install from source but No module named "mlx.core"
**Describe the bug**
mlx was build from source and installed successfully, but python said "No module named 'mlx.core'" and cannot generate stubs
```
Python 3.11.7 (main, Dec  4 2023, 18:10:11) [Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mlx.core as mlx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'mlx.core'
``` 

**To Reproduce**
1.  `git clone git@github.com:ml-explore/mlx.git mlx && cd mlx`
2.  `brew install pybind11`
3.  `env CMAKE_BUILD_PARALLEL_LEVEL="" pip install ".[dev]"`
4.  `python setup.py generate_stubs` and ModuleNotFoundError happened in this step

<details>
<summary> install log</summary>

```
…/mlx on  main via △ v3.28.1 via 🐍 v3.11.7

➜  env CMAKE_BUILD_PARALLEL_LEVEL="" pip install ".[dev]"
Processing /Users/user/Downloads/mlx
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: pre-commit in /opt/homebrew/lib/python3.11/site-packages (from mlx==0.0.9.dev2024114+6e81c3e) (3.6.0)
Collecting pybind11-stubgen (from mlx==0.0.9.dev2024114+6e81c3e)
  Using cached pybind11_stubgen-2.4.2-py3-none-any.whl.metadata (1.7 kB)
Requirement already satisfied: cfgv>=2.0.0 in /opt/homebrew/lib/python3.11/site-packages (from pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (3.4.0)
Requirement already satisfied: identify>=1.0.0 in /opt/homebrew/lib/python3.11/site-packages (from pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (2.5.33)
Requirement already satisfied: nodeenv>=0.11.1 in /opt/homebrew/lib/python3.11/site-packages (from pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (1.8.0)
Requirement already satisfied: pyyaml>=5.1 in /opt/homebrew/lib/python3.11/site-packages (from pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (6.0.1)
Requirement already satisfied: virtualenv>=20.10.0 in /opt/homebrew/lib/python3.11/site-packages (from pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (20.25.0)
Requirement already satisfied: setuptools in /opt/homebrew/lib/python3.11/site-packages (from nodeenv>=0.11.1->pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (69.0.2)
Requirement already satisfied: distlib<1,>=0.3.7 in /opt/homebrew/lib/python3.11/site-packages (from virtualenv>=20.10.0->pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (0.3.8)
Requirement already satisfied: filelock<4,>=3.12.2 in /opt/homebrew/lib/python3.11/site-packages (from virtualenv>=20.10.0->pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (3.13.1)
Requirement already satisfied: platformdirs<5,>=3.9.1 in /opt/homebrew/lib/python3.11/site-packages (from virtualenv>=20.10.0->pre-commit->mlx==0.0.9.dev2024114+6e81c3e) (4.1.0)
Using cached pybind11_stubgen-2.4.2-py3-none-any.whl (29 kB)
Building wheels for collected packages: mlx
  Building wheel for mlx (pyproject.toml) ... done
  Created wheel for mlx: filename=mlx-0.0.9.dev2024114+6e81c3e-cp311-cp311-macosx_14_0_arm64.whl size=11431839 sha256=18b83025e73176856c0f1c2c14766a91e4e3fcecc8acc3f84aad51bd692e4683
  Stored in directory: /private/var/folders/tw/7b3zd1w561z2m25dn0pz776h0000gn/T/pip-ephem-wheel-cache-cpt9tdq_/wheels/62/71/3d/0603f5068744a7bae1e68d1d256899ae698ad800182b546649
Successfully built mlx
Installing collected packages: pybind11-stubgen, mlx
Successfully installed mlx-0.0.9.dev2024114+6e81c3e pybind11-stubgen-2.4.2

```

</details>

**Expected behavior**
could import without error

**Desktop (please complete the following information):**
 - Apple M1 Max
 - OS Version: MacOS 14.2.1
 - Python Version: 3.11
 - Xcode Version: 15.2

### PR DESCRIPTION
**Pull request by awni**
## Proposed changes

See title.

Closes #454

### CODE REVIEW

**Comment by awni** [comment]
CC @dc-dc-dc should fix issues with stub generation, runs clean for me now.

File: setup.py [hunk 1]
```diff
@@ -135,18 +135,6 @@ def finalize_options(self):
 
     def run(self) -> None:
         subprocess.run(["pybind11-stubgen", "mlx.core", "-o", "python"])
-        # Note, sed inplace on macos requires a backup prefix, delete the file after its generated
-        # this sed is needed to replace references from py::cpp_function to a generic Callable
-        subprocess.run(
```
Comment (dc-dc-dc) [inline setup.py:140]: awesome, working localy!

**Comment by dc-dc-dc** [comment]
Sick, stubgen is working perfectly now 🚀

**Review by angeloskath — APPROVED** [review]
Very nice!

**Pull request merged**